src/Extractor.cc

```diff
@@ -2168,14 +2168,53 @@ namespace insur {
 	ril.index = discNumber;
 
 
-	//if (zmin > 0) {	
+	//if (zmin > 0) 	
         std::ostringstream dname, pconverter;
 	//disk name
         dname << xml_disc << discNumber; // e.g. Disc6
 
         std::map<int, ERingInfo> rinfo;
 	std::set<int> ridx;
 
+      
+        std::map<int, std::vector<double>> phi_one;
+        std::map<int, std::vector<double>> phi_two;
+        std::map<int, std::vector<double>> radius_one;
+        std::map<int, std::vector<double>> radius_two;
+        std::map<int, std::vector<double>> twist_one;
+        std::map<int, std::vector<double>> twist_two;
+
+        for (iiter = oiter->begin(); iiter != oiter->end(); iiter++){
+          if(iiter->getModule().uniRef().subdetectorName=="FPIX_1"){ //Don't need to check this for every endcap ring - awful hack, to be improved
```

Comment:
Yes please never use harcoded subdetector names. Otherwise the codebase become a CMSSW-style mess ;p
The only place where there are harcoded names is in the cabling (in a dedicated global string header), so that it is launched only on one given layout name (the cabling is layout-specific).

Here, you need to see what is the feature that makes the ring treatment different, and typically add a member variable in the ring hierarchy level, to acknowledge that.
You could add in the cfg files, when the irregualr mode is used, a bool:
isRegularRing false;
Which will set a boolean data member within the Ring class at CONSTRUCTION time. 
Then a getter in the Ring class:
isRegularRing() const { return isRegularRing_ }

Now, in the XML export, you can just call isRegularRing():
- if true, use the existing algo section;
- if false, build numModulesPerRing Pospart sections